src/build_queue.rs

```diff
@@ -188,6 +188,27 @@ impl BuildQueue {
     }
 }
 
+fn retry<T>(mut f: impl FnMut() -> Result<T>) -> Result<T> {
+    const MAX_ATTEMPTS: u8 = 3;
+    for attempt in 1.. {
+        match f() {
```

Comment:
this retries immediately, right? shouldn't we wait between requests?

Reply:
Valid point 😄

( funny I didn't think of that, must have been a case of nightly blackout 😂)

Reply:
I added exponential sleep between the tries